kernel/crates/kernel_vfs/src/path/absolute_owned.rs

```diff
@@ -74,3 +74,234 @@ impl AsRef<AbsolutePath> for AbsoluteOwnedPath {
+        let owned = OwnedPath::new("/");
+        let path: Result<AbsoluteOwnedPath, _> = owned.try_into();
+        assert!(path.is_ok());
+        assert_eq!(path.unwrap().as_str(), "/");
+
+        let owned = OwnedPath::new("/foo/bar");
+        let path: Result<AbsoluteOwnedPath, _> = owned.try_into();
+        assert!(path.is_ok());
+        assert_eq!(path.unwrap().as_str(), "/foo/bar");
+    }
+
+    #[test]
+    fn test_try_from_owned_path_invalid() {
+        let owned = OwnedPath::new("");
+        let path: Result<AbsoluteOwnedPath, _> = owned.try_into();
+        assert!(path.is_err());
+
+        let owned = OwnedPath::new("foo");
+        let path: Result<AbsoluteOwnedPath, _> = owned.try_into();
+        assert!(path.is_err());
+
+        let owned = OwnedPath::new("foo/bar");
+        let path: Result<AbsoluteOwnedPath, _> = owned.try_into();
+        assert!(path.is_err());
+    }
+
+    #[test]
+    fn test_deref() {
+        let abs_path = AbsoluteOwnedPath::new();
+        let owned: &OwnedPath = &*abs_path;
+        assert_eq!(owned.as_str(), "/");
+
+        // Should have access to OwnedPath methods
+        assert!(abs_path.is_absolute());
+        assert_eq!(abs_path.file_name(), None);
+    }
+
+    #[test]
+    fn test_deref_mut() {
+        let mut abs_path = AbsoluteOwnedPath::new();
+        abs_path.push("foo");
+        assert_eq!(abs_path.as_str(), "/foo");
+
+        abs_path.push("bar");
+        assert_eq!(abs_path.as_str(), "/foo/bar");
+
+        abs_path.append_str(".txt");
+        assert_eq!(abs_path.as_str(), "/foo/bar.txt");
+    }
+
+    #[test]
+    fn test_as_ref_absolute_path() {
+        let abs_owned = AbsoluteOwnedPath::new();
+        let abs_path: &AbsolutePath = abs_owned.as_ref();
+        assert_eq!(&abs_path.to_string(), "/");
+    }
+
+    #[test]
+    fn test_borrow() {
+        use core::borrow::Borrow;
+
+        let abs_owned: AbsoluteOwnedPath = "/foo/bar".try_into().unwrap();
+        let borrowed: &AbsolutePath = abs_owned.borrow();
+        assert_eq!(&borrowed.to_string(), "/foo/bar");
+    }
+
+    #[test]
+    fn test_clone() {
+        let path: AbsoluteOwnedPath = "/foo/bar".try_into().unwrap();
+        let cloned = path.clone();
+        assert_eq!(path, cloned);
+        assert_eq!(cloned.as_str(), "/foo/bar");
+    }
+
+    #[test]
+    fn test_eq() {
+        let path1: AbsoluteOwnedPath = "/foo/bar".try_into().unwrap();
+        let path2: AbsoluteOwnedPath = "/foo/bar".try_into().unwrap();
+        let path3: AbsoluteOwnedPath = "/foo/baz".try_into().unwrap();
+
+        assert_eq!(path1, path2);
+        assert_ne!(path1, path3);
+    }
+
+    #[test]
+    fn test_ord() {
+        let path1: AbsoluteOwnedPath = "/a".try_into().unwrap();
+        let path2: AbsoluteOwnedPath = "/b".try_into().unwrap();
+        let path3: AbsoluteOwnedPath = "/c".try_into().unwrap();
+
+        assert!(path1 < path2);
+        assert!(path2 < path3);
+        assert!(path1 < path3);
```

Comment:
I don't want the order guaranteed, I just want the fact that the paths are `Ord` so that I can use them in a `BTreeMap` or similar. I want to be free to change the order without having to adapt such tests. Thus, please remove this test.